apollo-backend/src/test/java/io/logz/apollo/DeploymentTest.java

```diff
@@ -65,14 +76,37 @@ public void testGetAllDeployments() throws Exception {
     @Test
     public void testSimultaneousDeployments() throws Exception {
 
-        ApolloTestClient apolloTestClient = Common.signupAndLogin();
-
         Deployment deployment1 = createAndSubmitDeployment(apolloTestClient);
 
         // Submit that again to verify we can't run the same one twice
-        assertThatThrownBy(() -> apolloTestClient.addDeployment(deployment1)).isInstanceOf(ApolloClientException.class);
+        assertThat(apolloTestClient.addDeployment(deployment1).getUnsuccessful().size()).isEqualTo(1);
+        assertThat(apolloTestClient.addDeployment(deployment1).getSuccessful().size()).isEqualTo(0);
 
         // Just to make sure we are not blocking different deployments to run on the same time
         createAndSubmitDeployment(apolloTestClient);
     }
+
+    @Test
+    public void testMultiServiceDeployment() throws Exception {
+        Environment env1 = createAndSubmitEnvironment(apolloTestClient);
+        Environment env2 = createAndSubmitEnvironment(apolloTestClient);
+        Service service1 = createAndSubmitService(apolloTestClient);
+        Service service2 = createAndSubmitService(apolloTestClient);
+        DeployableVersion deployableVersion1 = createAndSubmitDeployableVersion(apolloTestClient, service1);
+        createAndSubmitDeployableVersion(apolloTestClient, service2, deployableVersion1.getGithubRepositoryUrl(), deployableVersion1.getGitCommitSha());
+
+        String envIdsCsv = String.valueOf(env1.getId()) + "," + String.valueOf(env2.getId());
+        String serviceIdsCsv = String.valueOf(service1.getId()) + "," + String.valueOf(service2.getId());
+
+        ModelsGenerator.createAndSubmitPermissions(apolloTestClient, Optional.of(env1), Optional.empty(), DeploymentPermission.PermissionType.ALLOW);
+        ModelsGenerator.createAndSubmitPermissions(apolloTestClient, Optional.of(env2), Optional.empty(), DeploymentPermission.PermissionType.ALLOW);
+
+        MultiDeploymentResponseObject result = apolloTestClient.addDeploymentWithParams(envIdsCsv, serviceIdsCsv, deployableVersion1.getId());
+
+        assertThat(result.getSuccessful().size()).isEqualTo(4);
+        assertThat(result.getUnsuccessful().size()).isEqualTo(0);
+
+        LinkedHashMap deployment = (LinkedHashMap) result.getSuccessful().get(0).get("deployment");
+        assertThat(deploymentDao.getDeployment((int) deployment.get("id"))).isNotNull();
```

Comment:
Why using the Dao? Just get it from the API

Try not to use DAOs as much as possible when testing